fvtest/tril/tril/enable_if.hpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+#ifndef ENABLE_IF_HPP
+#define ENABLE_IF_HPP
+
+#if __cplusplus > 199711L
```

Comment:
This version check seems somewhat aggressive to me, I would have thought we would want something more along the lines of
```
#if (__cplusplus < 201103L) || !defined(_MSVC_LANG)
```
with `_MSVC_LANG` assuming that MSVC++2015+ is acceptable...

Do we know for sure that all partial updates to the C++ language standard on the compilers we care about either
1. Don't update the _cplusplus version or,
2. Are compatible with our requirements?

Reply:
@lmaisons 

This was brought up in an earlier comment, but I don't think there's any way of coming up with a combination of macros that is guaranteed to be infallible when trying to detect whether the compiler does or doesn't have C++11 support [checking `__cplusplus` by itself *theoretically should be enough] . 

>"Implementations rarely leap from one formal revision of the standard directly to the next; the implementation process generally proceeds by smaller steps. As a result, testing for a specific revision of the standard (e.g. by examining the value of the __cplusplus macro) often gives the wrong answer. Implementers generally don’t want to appear to be claiming full conformance to a standard revision until all of its features are implemented. That leaves programmers with no portable way to determine which features are actually available to them."

 [ ref- https://isocpp.org/std/standing-documents/sd-6-sg10-feature-test-recommendations#recs.hasinc . I guess C++17 has `__has_cpp_attribute` which kind of does the trick, but of course that's no use to us.  ]

Right now how it's set up, it at least works for our build compiler and the newest GCC one. Not sure how much marginal benefit we might potentially be able to extract from complicating the `#if` statements and whether it's even "worth" doing that. I might be missing something here.